orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/DetermineHealthProvidersTask.java

```diff
@@ -83,6 +88,18 @@ public TaskResult execute(Stage stage) {
       }
 
       Application application = front50Service.get(applicationName);
+      List<String> processedInterestingHealthProviderNames = interestingHealthProviderNamesProcessors
+        .stream()
+        .filter(healthProcessor -> healthProcessor.supports(getCloudProvider(stage), application, stage))
+        .flatMap(healthProcessor -> healthProcessor.process(stage).stream())
```

Comment:
What about if we filtered it down and took the first match ... it's probably a little confusing to process multiple times. I'm not sure there would ever be a legitimate reason to do so.